src/Ide/Types.hs

```diff
@@ -169,7 +169,8 @@ data FormattingType = FormatText
 -- It is required to pass in the whole Document Text for that to happen, an empty text
 -- and file uri, does not suffice.
 type FormattingProvider m
-        = IdeState
+        = LSP.LspFuncs Config
+        -> IdeState
```

Comment:
This change looks innocent, but does break a layer?
Other providers have access to the lspfuncs, too.

Reply:
I meant the original commented-out code dived directly into the bios, instead of working with lspfuncs.